indexd/index/drivers/alchemy.py

```diff
@@ -801,6 +842,15 @@ def update(self, did, rev, changing_fields):
                     for ace in set(changing_fields['acl'])
                 ]
 
+            if 'authz' in changing_fields:
+                for resource in record.authz:
+                    session.delete(resource)
```

Comment:
I assume `commit()` gets called somewhere outside this function?

Reply:
Yes, `with self.session as session` does that I believe.